### DISCUSSION TITLE
Any way to have the label auto-size to fit the whitespace provided by the splice?

### DISCUSSION THREAD

**Discussion by trustieee** [ImageMagick/ImageMagick#5611] [Help] [answered]
I'm finding I have to play with the `fx:w*0.1` number to make the label fit inside the whitespace provided by the splice before it. Is there a better way to do this? Or to auto size it?

```
magick tiff1.tiff -splice 0x2.5% -size %[fx:w*0.1]x ( label:"FOOBAR" ) -composite OUT/tiff1_magick2.tiff
```
Example of what I'm getting now, which isn't right:
![image](https://user-images.githubusercontent.com/1885172/193394175-60cd00c8-8ba3-4bc3-b9f7-d1d27ac31e0c.png)

**Comment by GeeMack** [comment, marked as answer]
You're setting the height of the splice according to the input image, and you're setting the width of the label relative to the input image. If you use a relative height dimension for both it should fit...

    magick -size x%[fx:h*0.025] -splice 0x2.5% label:"TEXT" ...

That sets both the "-size" height for the label and the height of the splice according to the dimension before the splice is added.
